cbv3_django_prototype/cbv3_django_prototype/tagging/migrations/0001_initial.py

```diff
@@ -0,0 +1,30 @@
+# Generated by Django 2.2.4 on 2020-02-29 23:16
+
+from django.db import migrations, models
+import uuid
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = [
+        ('taggit', '0003_taggeditem_add_unique_index'),
```

Comment:
Elementary question: what does this `0003` number represent?

Reply:
(It's auto-generated by Django, but I wonder why I'm seeing `0004_taggeditems` elsewhere. Maybe this is not an important detail.

Reply:
The numbers are given so that migrations can be executed in the order they were developed. They are auto incrementing and this is why I would never expect to see modifications to migrations.

Reply:
The numbers are auto-assigned by Django as the ORM generates the changes needed for the database.  They're generated/added to each time **`makemigrations`** is run **for the system you are on**.  

![image](https://user-images.githubusercontent.com/5923094/75628230-d4621180-5b8b-11ea-98ce-d13a9f2b75c0.png)

...migrations are not run in the order they are developed -- the dependency tree is resolved **at runtime** by the Django ORM.  This usually works fine in the case of straightforward additive changes...and breaks entirely with things like moving a manytomany relationship around without specifying an ordering, or needed to override how a third party app is behaving (like our situation).  

![image](https://user-images.githubusercontent.com/5923094/75628309-78e45380-5b8c-11ea-80c3-6062b02a644c.png)

Please see [**controlling the order of Migrations**](https://docs.djangoproject.com/en/3.0/howto/writing-migrations/#controlling-the-order-of-migrations) for more info.

The big problem here is that these auto-generated and auto-numbered files are done when **`makemigrations`** is run .... and we don't have any control over how Django does that for third party Django apps, so we don't have control over what numbers get assigned to **taggit**s migration files on install.

I am going to see if just specifying the **0001** file forces the correct order.  We know, at least that **0001** will always be generated.

Reply:
_**I spoke too soon**_  See [**getting the last migration**](https://stackoverflow.com/questions/29416956/how-to-get-the-last-migration) for more info.  Will push a fix shortly.

Reply:
Apologies.  To also answer the earlier question:
 **'0003_taggeditem_add_unique_index'** is the third migration file for **`taggit`**.  

The **`0004_taggeditem`** references are for the 4th migration file for the **tagging** app, which inherits from, and is dependent on, **taggit**.